crates/oxc_parser/src/js/expression.rs

```diff
@@ -451,13 +451,17 @@ impl<'a> ParserImpl<'a> {
     ///     `SubstitutionTemplate`[?Yield, ?Await, ?Tagged]
     pub(crate) fn parse_template_literal(&mut self, tagged: bool) -> TemplateLiteral<'a> {
         let span = self.start_span();
-        let mut expressions = self.ast.vec();
-        let mut quasis = self.ast.vec();
-        match self.cur_kind() {
+
+        let (quasis, expressions) = match self.cur_kind() {
             Kind::NoSubstitutionTemplate => {
+                let mut quasis = self.ast.vec_with_capacity(1);
                 quasis.push(self.parse_template_element(tagged));
+                (quasis, self.ast.vec())
             }
             Kind::TemplateHead => {
+                let mut expressions = self.ast.vec_with_capacity(1);
+                let mut quasis = self.ast.vec_with_capacity(2);
```

Comment:
we could be a bit more liberal here and do 4 expressions + 5 quasis at the cost of empty vec elements/mem usage 🤷

Reply:
Really hard to know what the best values for pre-allocation are. Over-allocating often makes parser a bit faster, but everything else that consumes the AST slower. I would tend to think 1 / 2 is probably not a bad default, but who knows? It depends on the code, and no way to know ahead of time.

Reply:
yeah 1/2 is the absolute minimum count that it could be here.

```js
`first_quasis${ first_expresison }second_quasis`
```

even though there's negilgble perf differences shall we merge this like it anyway?